### PR TITLE
Add make package target

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by bac** [comment]
With the addition of the 'make package' target, the version of the jujugui-charm that we keep in version control no longer has all of the dependencies required by the juju gui and doesn't have the package for the gui.  Instead, running `make package` will fetch the latest repo of juju gui from github and build from the master branch, copying the tarball and all dependencies into place.  Once done, `charm2 upload` can be used and the fat charm will be uploaded with the temporary files (`releases/jujugui-<latest>.bz2` and `jujugui-deps`) uploaded.

This approach is a spike to see if it is a better way to manage fat charms. There are fewer duplications across codebases, so that's a good thing.  Publishing a new charm release becomes simply:
```
make package
<test, test, test>
charm2 upload . cs:~bac/trusty/jujugui
charm2 change-perm --add-read=everyone cs:~bac/trusty/jujugui
charm2 publish cs:~bac/trusty/jujugui
```

QA:
First, set environment variables to talk to jujugui.org so the real charmstore isn't polluted.
`export JUJU_CHARMSTORE=http://api.jujugui.org/charmstore`

```
make package
make deploy
```

You can then try uploading and publishing the charm and deploying from jujugui.org.

File: src/Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+JUJU_GUI_PROJECT ?= https://github.com/juju/juju-gui.git
+
+juju-gui:
+	git clone $(JUJU_GUI_PROJECT)
```
Comment (hatched) [inline src/Makefile]: this doesn't need to clone the entire repo, just the latest level, it should probably also define the branch

File: src/Makefile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+.PHONY: package
+package: juju-gui
+	cd juju-gui && \
+	  git pull && \
```
Comment (hatched) [inline src/Makefile]: if you take into account what I said above, I'm not sure what this will do... :)

File: src/Makefile (hunk 1 above)
Comment (hatched) [inline src/Makefile]: I think you want

git clone --depth 1 --branch master https://github.com/juju/juju-gui.git

Comment (bac) [inline src/Makefile]: That looks like a great improvement. Thanks.

File: src/Makefile (hunk 2 above)
Comment (hatched) [inline src/Makefile]: With the above change I believe the proper syntax for this is

`git pull https://github.com/juju/juju-gui.git master`

**Comment by makyo** [comment]
Sweet, +1!

File: src/Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+JUJU_GUI_PROJECT ?= https://github.com/juju/juju-gui.git
+
+juju-gui:
+	git clone --depth 1 --branch master $(JUJU_GUI_PROJECT)
```
Comment (jaycee) [inline src/Makefile]: Are we copying release versions over to master from development? I thought we weren't doing that yet.

**Comment by jaycee** [comment]
:+1: though I am surprised to find that we are evidently copying releases to master. I thought we weren't doing that yet for some reason.

Mind you, I'm pleased that we are.

**Comment by jrwren** [comment]
Things seemed to work nicely until one error killed it all.

    no previously-included directories found matching 'docs/_build'
Successfully downloaded Mako MarkupSafe Paste PasteDeploy Pygments WebOb argparse convoy pyramid-mako pyramid repoze.lru six translationstring venusian waitress wsgiref zope.deprecation zope.interface
Cleaning up...
make[2]: Leaving directory '/home/jrwren/src/charms/juju-gui/src/juju-gui'
make[1]: Leaving directory '/home/jrwren/src/charms/juju-gui/src'
cp -r src/juju-gui/dist/* releases
cp -r src/juju-gui/collected-requirements/* jujugui-deps
cp: target ‘jujugui-deps’ is not a directory
Makefile:78: recipe for target 'package' failed
make: *** [package] Error 1

**Comment by jrwren** [comment]
:+1:  QA looks good

**Comment by bac** [comment]
:shipit:

**Pull request merged**